edb/repl/table.py

```diff
@@ -0,0 +1,170 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2019-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from __future__ import annotations
+from typing import *  # NoQA
+
+import math
+import sys
+import textwrap
+import typing  # needed for linting TextIO
+
+import edgedb
+
+
+class ColumnSpec(NamedTuple):
+    field: str
+    title: str
+    width: int
+    align: str
+
+
+def render_table(
+    title: str,
+    columns: List[ColumnSpec],
+    data: Iterable[edgedb.Object],
+    max_width: int = 0,
+    *,
+    file: typing.TextIO = sys.stdout,
+) -> None:
+    if max_width == 0:
+        return
+
+    # find the amount of space available for text
+    avail_width = max(0, max_width - (3 * len(columns) - 1))
```

Comment:
Pls add a comment to explain this formula

Reply:
Comment added.